src/Microsoft.AspNet.SignalR.Client/Connection.cs

```diff
@@ -511,35 +511,40 @@ Task StartNegotiation()
                                 {
                                     VerifyProtocolVersion(negotiationResponse.ProtocolVersion);
 
-                                    if (negotiationResponse.ProtocolVersion.Equals("2.0") && !string.IsNullOrEmpty(negotiationResponse.RedirectUrl))
+                                    if (negotiationResponse.ProtocolVersion.Equals("2.0"))
                                     {
-                                        // Ensure the URL ends with a trailing slash.
-                                        if (!negotiationResponse.RedirectUrl.EndsWith("/"))
+                                        if (!string.IsNullOrEmpty(negotiationResponse.Error))
                                         {
-                                            negotiationResponse.RedirectUrl += "/";
+                                            // REVIEW: Exception type?
+                                            throw new Exception(string.Format(Resources.Error_ErrorFromServer, negotiationResponse.Error));
```

Comment:
Hmm. Would `ExternalException` work here? I've never actually used it but found it in a quick search

Reply:
That exception is for COM Interop. We generally avoid throwing those.